toolkit/tools/pkg/imagecustomizerlib/kernelmoduleutils.go

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+)
+
+type LoadMode string
+
+const (
+	LoadModeBoot    LoadMode = "boot"
+	LoadModeAuto    LoadMode = "auto"
+	LoadModeDisable LoadMode = "disable"
+	LoadModeInherit LoadMode = "inherit"
+)
+
+func loadOrDisableModules(modules []imagecustomizerapi.Module, imageChroot *safechroot.Chroot) error {
```

Comment:
Considering that this function only uses `imageChroot.RootDir()`, it might be goodness to remove the `imageChroot` parameter and pass the rootDir as a string instead. This would make it easier to write tests for this function.